### PR TITLE
feat(contact-center): fixed-dest-agent-id-obtaining

### PR DESCRIPTION
**Pull request by adhmenon**
# COMPLETES #< AD-HOC>

## This pull request addresses

* In the previous PR to ensure consultTransfer to dial number works, we had removed a task.updateTaskData for the CONSULTING event.
* This caused an issue on widgets as now for receiving consult, we were looking at some value inside that updated data for UI changes.
* Hence e2e tests were failing
* Now, we also were incorrectly reading the `destAgentId` from the task data - agent desktop has a different method.

## by making the following changes

* Fixed the consult UI issue by re-adding the update task data.
* We are now extracting the `destAgentId` from the task data but in the same way agent desktop does.
* The logic is a bit weird but they simply use the `this.data.interactions.participants` object and extratc the id of the non-user agent there
* Added unit tests for the same

E2E Tests for Consult passing in widgets - 

Screenshot 2025-09-16 at 12 39 54 PM

Screenshot 2025-09-16 at 2 20 48 PM

Widgets Vidcast showing consult transfer working - https://app.vidcast.io/share/2562d51c-3e3c-41fc-9960-50c8c8d55913

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

* Ran e2e tests on widgets locally
* Basic call control tests

## The GAI Coding Policy And Copyright Annotation Best Practices ##
 

 
- [ ] GAI was not used (or, no additional notation is required)
- [ ] Code was generated entirely by GAI
- [x] GAI was used to create a draft that was subsequently customized or modified
- [ ] Coder created a draft manually that was non-substantively modified by GAI (e.g., refactoring was performed by GAI on manually written code)
- [ ] Tool used for AI assistance (GitHub Copilot / Other - specify)
  - [ ] Github Copilot
  - [x] Cursor
- [ ] This PR is related to
  - [x] Feature
  - [ ] Defect fix
  - [ ] Tech Debt
  - [ ] Automation

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request
- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/contact-center/src/services/core/Utils.ts [hunk 1]
```diff
@@ -166,6 +171,50 @@ export const createErrDetailsObject = (errObj: WebexRequestPayload) => {
  * @param taskData - The task data used to infer the agent action and destination type
  * @returns The normalized destination type to be used for consult transfer
  */
+/**
+ * Checks if a participant type represents a non-customer participant.
+ * Non-customer participants include agents, dial numbers, entry point dial numbers,
+ * and entry points.
+ */
+const isNonCustomerParticipant = (participantType: string): boolean => {
```
Comment (adhmenon) [inline packages/@webex/contact-center/src/services/core/Utils.ts:179]: Have added these helper methods... same as agent desktop

File: packages/@webex/contact-center/src/services/core/Utils.ts [hunk 2]
```diff
@@ -166,6 +171,50 @@ export const createErrDetailsObject = (errObj: WebexRequestPayload) => {
...
+
+/**
+ * Gets the destination agent ID from participants data by finding the first
+ * non-customer participant that is not the current agent and is not in wrap-up state.
+ *
+ * @param participants - The participants data from the interaction
+ * @param agentId - The current agent's ID to exclude from the search
+ * @returns The destination agent ID, or empty string if none found
+ */
+export const getDestinationAgentId = (
```
Comment (adhmenon) [inline packages/@webex/contact-center/src/services/core/Utils.ts:196]: As mentioned, in their logic they simply check the `participants` list and based on the `id`, eliminate the current `agent` and use only the consulting agent. 
Have verified that this works cleanly with widgets in the vidcast.

File: packages/@webex/contact-center/src/services/task/index.ts [hunk 3]
```diff
@@ -1310,13 +1314,19 @@ export default class Task extends EventEmitter implements ITask {
     consultTransferPayload?: ConsultTransferPayLoad
   ): Promise<TaskResponse> {
     try {
+      // Get the destination agent ID using custom logic from participants data
+      const destAgentId = getDestinationAgentId(
```
Comment (adhmenon) [inline packages/@webex/contact-center/src/services/task/index.ts:1318]: Let us just use this Utils method... we will remove the other payload and signature as part of another ticket. Don't want to break the docs and webex one demos.

File: packages/@webex/contact-center/test/unit/spec/services/core/Utils.ts [hunk 4]
```diff
@@ -276,4 +276,54 @@ describe('Utils', () => {
       });
     });
   });
+
+  describe('getDestinationAgentId', () => {
```
Comment (adhmenon) [inline packages/@webex/contact-center/test/unit/spec/services/core/Utils.ts:280]: We are solidly testing all edge cases of the extraction method.

File: packages/@webex/contact-center/test/unit/spec/services/task/index.ts [hunk 5]
```diff
@@ -141,6 +142,11 @@ describe('Task', () => {
       },
     };
 
+    // Mock destination agent id resolution from participants
+    getDestinationAgentIdSpy = jest
```
Comment (adhmenon) [inline packages/@webex/contact-center/test/unit/spec/services/task/index.ts:146]: Mocking this for this particular unit test since we have solidly tested it in the Utils test cases.

**Review by rsarika — APPROVED** [review]

**Review by Shreyas281299 — APPROVED** [review]

**Comment by adhmenon** [comment]
**Note** 
Had a manual testing session with Ravi. 
All cases were passing - we were unable to test consult-transfer to queue with dial number as we could not figure out how to set it up. However, it should still work as the logic for consult-transfer to dial number and consult-transfer to queue with dial number is ideally the same.
